src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.ts

```diff
@@ -91,39 +92,53 @@ export class ParticipationSubmissionComponent implements OnInit {
-                        solutionParticipation.programmingExercise = this.exercise;
-                    } else {
-                        // Should not happen
-                        alert(this.translateService.instant('artemisApp.participation.noParticipation'));
-                    }
-                    this.isLoading = false;
-                });
-            } else {
-                // Get exercise for release and due dates
-                this.exerciseService.find(params['exerciseId']).subscribe((exerciseResponse) => {
-                    this.exercise = exerciseResponse.body!;
-                    this.updateStatusBadgeColor();
-                });
-                this.fetchParticipationAndSubmissionsForStudent();
-            }
+                        // Check if requested participationId belongs to the template or solution participation
+                        if (this.participationId === templateParticipation?.id) {
+                            this.participation = templateParticipation;
+                            this.submissions = templateParticipation.submissions!;
+                            // This is needed to access the exercise in the result details
+                            templateParticipation.programmingExercise = this.exercise;
+                        } else if (this.participationId === solutionParticipation?.id) {
+                            this.participation = solutionParticipation;
+                            this.submissions = solutionParticipation.submissions!;
+                            // This is needed to access the exercise in the result details
+                            solutionParticipation.programmingExercise = this.exercise;
+                        } else {
+                            // Should not happen
+                            alert(this.translateService.instant('artemisApp.participation.noParticipation'));
+                        }
+
+                        if (this.submissions) {
+                            this.submissions.forEach((submission: ProgrammingSubmission) => {
+                                if (submission.results) {
+                                    submission.results.forEach((result: Result) => {
+                                        result.logsAvailable = this.logsAvailable?.[result.id!];
+                                    });
+                                }
+                            });
+                        }
+
+                        this.isLoading = false;
+                    });
+                } else {
+                    // Get exercise for release and due dates
+                    this.exerciseService.find(params['exerciseId']).subscribe((exerciseResponse) => {
+                        this.exercise = exerciseResponse.body!;
+                        this.updateStatusBadgeColor();
+                    });
+                    this.fetchParticipationAndSubmissionsForStudent();
```

Comment:
does the else-case also need a `this.isLoading = false`?

Reply:
good point